assets/js/Common/EventSource.js

```diff
@@ -42,7 +46,15 @@ customElements.define('event-source', class extends HTMLElement {
         if (this._verbose) console.log(eventName, payload);
     }
 
-    _onAddSubscription(event) {
+    _onError = async () => {
+        if (this._eventSource.readyState !== EventSource.CLOSED) return;
+
+        const timeout = 4000 + Math.floor(Math.random() * 2000);
+        window.app.notifyUser('No connection to server.', 'danger', timeout);
```

Comment:
Instead of notifying the user every time, they should only be notified once. Further, there can be a new online/offline indicator which is updated in case of a new connection status. Without an online/offline indicator, the notification could be shown permanently, but then it should be dismissible.

Alternatively, there can be another global notification that is integrated into the page flow but is fixed, just like the header.